components/arm/collectors.go

```diff
@@ -32,8 +33,13 @@ func newEndPositionCollector(resource interface{}, params data.CollectorParams)
 	}
 
 	cFunc := data.CaptureFunc(func(ctx context.Context, _ map[string]*anypb.Any) (interface{}, error) {
+		ctx = context.WithValue(ctx, data.FromDMContextKey{}, true)
 		v, err := arm.EndPosition(ctx, nil)
 		if err != nil {
+			// If err is from a modular filter component, propagate it to getAndPushNextReading().
```

Comment:
I do think for someone working in RDK, this is a little confusing. They may be wondering what a modular filter component is. I think it may be worth adding a note: "An modular filter component can be created to filter the readings from a component. The error data.ErrNoCapture is used in the datamanager to exclude readings from being captured". I know it's long... but I think I would be confused if I was working in this code base and I saw this suddenly pop up in the capture function.

Reply:
Definitely - adding this to all the sites 👍